packages/gafl-webapp-service/src/handlers/multibuy-duplicate-handler.js

```diff
@@ -0,0 +1,23 @@
+import _ from 'lodash'
+
+export const checkDuplicates = async licences => {
+  // break down parts of licence to check for duplication
+  const licencesCheck = licences.map(licences => ({
+    licenceHolder: licences.licenceHolder,
+    type: licences.type,
+    length: licences.length,
+    start: licences.start
+  }))
+
+  let checkDuplicates = false
+  for (let licence = 0; licence < licencesCheck.length; licence++) {
+    for (let licenceCompare = licence + 1; licenceCompare < licencesCheck.length; licenceCompare++) {
+      if (_.isEqual(licencesCheck[licence], licencesCheck[licenceCompare])) {
+        checkDuplicates = true
+        break
+      }
+    }
+  }
```

Comment:
I think this is a bit complex. Essentially, you need a hash that allows for quick comparison between elements. If it's an identical licenceHolder, type, length and start date that constitute a duplicate, you can create a 'comparison array', by doing:
```
const hasDuplicates = licencesArray => {
  const haystack = licencesArray.map(licence => `${licence.licenceHolder}${licence.type}${licence.length}${licence.start}`);
  const duplicates = haystack.filter(needle => { 
      return haystack.filter(potentialNeedle => needle === potentialNeedle).length > 1
  });
  return duplicates.length > 1
}
```
There are a few other good examples here: https://stackoverflow.com/questions/30735465/how-can-i-check-if-the-array-of-objects-have-duplicate-property-values, I think using a Set might be a good option although I'm not sure that it's supported in the version of Node we're on:
```
let seen = new Set();
var hasDuplicates = values.some(function(currentObject) {
    return seen.size === seen.add(currentObject.name).size;
});
```